driver/csiplugin/gpfs_util.go

```diff
@@ -264,6 +283,34 @@ func getScaleVolumeOptions(volOptions map[string]string) (*scaleVolume, error) {
 		scaleVol.NodeClass = nodeClass
 	}
 
+	if isCompressionSpecified {
+		if !IsValidCompressionAlgorithm(compression) {
+			glog.V(5).Infof("gpfs_util invalid compression algorithm specified: %s",
+				compression)
+			return &scaleVolume{}, status.Errorf(codes.InvalidArgument,
+				"invalid compression algorithm specified: %s", compression)
+		}
+		scaleVol.Compression = compression
+		glog.V(5).Infof("gpfs_util compression was set to %s", compression)
+	}
+
+	if isReplicationSpecified && replication != "" {
+		val, err := strconv.Atoi(replication)
+		if err != nil || val < 1 || val > 3 {
+			glog.V(5).Infof("gpfs_util invalid replication value specified: %s",
+				replication)
+			return &scaleVolume{}, status.Errorf(codes.InvalidArgument,
+				"invalid replication value specified: %s", replication)
+		}
+		scaleVol.Replication = val
+		glog.V(5).Infof("gpfs_util replication was set: %d", val)
+	}
+
+	if isTierSpecified && tier != "" {
```

Comment:
If the tier/pool does not exists, mmchpolicy will fail, but maybe we could catch this earlier? not a big deal if we don't as long as we fail the creation?

Reply:
This gets called from the CreateVolume so it will fail shortly after this regardless. After looking at this, I am wondering if we want to set the policy before or after we have a volume created though. If it's before and the volume creation fails we don't do any cleanup of the policy. We may want to run SetFilesystemPolicy after a successful CreateVolumeResponse or figure out a way to do some cleanup. @deeghuge thoughts?

Reply:
I think we should first check if pool exist before fileset is created. 
Actual policy creation can be done after creating the fileset. 
Reason being if create the fileset before and later found out that pool not in place and we fail at this point if user decided to delete the pvc then we won't be able to do cleanup of fileset. 

We usually try to do prerequisite check before creating the actual object on scale

Reply:
Added a check using /scalemgmt/v2/filesystems/{filesystemName}/pools/{storagePoolName} to make sure the storage pool is valid. If it is not, the user will receive a message like:
```
  Warning  ProvisioningFailed    0s (x5 over 13s)         spectrumscale.csi.ibm.com_ibm-spectrum-scale-csi-provisioner-0_c091c801-eb56-47d3-a163-00d98f937663  failed to provision volume with StorageClass "ibm-spectrum-scale-compression": rpc error: code = Unknown desc = invalid storagePool 'FAKE' specified for filesystem fs0
```